améliorer la détection des urls identiques

brunob opened the pull request:
lors de l'import d'un flux matcher les urls en http/https comme identiques / ref https://github.com/seenthis/seenthis_squelettes/issues/130

should fix seenthis/seenthis_squelettes#216

brunob committed:
améliorer la détection des urls identiques

lors de l'import d'un flux matcher les urls en http/https comme identiques / ref https://github.com/seenthis/seenthis_squelettes/issues/130

Fil commented on genie/seenthis_importer_flux.php:
```diff
@@ -89,9 +89,10 @@ function seenthis_importer_rss_article($article, $moi, $create=true) {
 			$q = 'SELECT t.id_me,m.id_auteur
 			FROM spip_me_tags AS t
-			INNER JOIN spip_me AS m ON t.uuid=m.uuid AND t.tag='.sql_quote($url);
+			INNER JOIN spip_me AS m ON t.uuid=m.uuid AND t.tag LIKE '.sql_quote('%' . $lien_flou);
```
ça risque de matcher sur d'autres trucs, par exemple de dire que toto.rezo.net est LIKE rezo.net

Fil replied:
👍

brunob committed:
simplification et bugfix pour la détection http/https + fix syntax error

utliser un simple OR plutôt qu'un like évasif qui remontait trop de choses (merci fil)

Fil commented on genie/seenthis_importer_flux.php:
```diff
@@ -89,9 +89,10 @@ function seenthis_importer_rss_article($article, $moi, $create=true) {
 			# qu'il ne reviennent bégayer...
 			# 2. si un lien existe et appartient à quelqu'un d'autre,
 			# on le partage, sauf si on a bloqué la personne
+			$lien_flou = preg_replace(',^(https?://)?,i', '', $url);
```
le ? ne sert pas :)

brunob replied:
Ha re merdum, s'il ne sert pas ici il ne sert nul part dans les autres patchs associés alors ?

Fil replied:
C'est vraiment un détail, ça ne gêne pas non plus… S'il n'y a pas de match sur https:// ou http://, on remplace "" par "". Sans la parenthèse, on ne remplacerait pas. Ca ferait juste un code un peu plus court:

> preg_replace(',^https?://,i', '', $url);

bref aucune importance, le mieux est d'avoir la même regexp dans les trois endroits

brunob replied:
Les détails comptent :) Mais disons qu'on va merger et qu'il faudra y revenir afin de modifier la regexp partout, thx again :)
